packages/ai-jsx/src/experimental/next/chat.tsx

```diff
@@ -0,0 +1,110 @@
+/** @jsxImportSource ai-jsx/react */
+import 'server-only';
+import { AssistantMessage, UserMessage } from '../../core/completion.js';
+import {
+  MessageMap,
+  Message as ClientMessage,
+  Conversation as ClientConversation,
+  SendMessage,
+} from './chat.client.js';
+export { Chat, ChatState } from './chat.client.js';
+import * as AI from './index.js';
+
+const AIMessageContext = AI.createContext(null as AI.Node);
+const AIConversationContext = AI.createContext([] as SerializedMessage[]);
+
+/**
+ * A serialized chat message.
+ */
+export interface SerializedMessage {
+  type: 'user' | 'assistant';
+  message: string;
+}
+
+/**
+ * A component that will resolve to a single message in a conversation. Used as part of
+ * the `children` prop of `<Conversation>`.
+ */
+export function Message(props: {}, maybeContext?: AI.ComponentContext) {
+  if (maybeContext && 'render' in maybeContext) {
+    return maybeContext.getContext(AIMessageContext);
+  }
+
+  return <ClientMessage />;
+}
+
+/**
+ * A component that will resolve to the conversation so far. Can be used
+ * within either AI.JSX or React. The children prop defines how each type
+ * of message will be presented.
+ *
+ * By default in AI.JSX the conversation will resolve to a format that
+ * can be provided to <ChatCompletion>.
+ */
+export function Conversation(
+  {
+    children,
+  }: {
+    children?: MessageMap;
+  },
+  maybeContext?: AI.ComponentContext
+) {
+  if (maybeContext && 'render' in maybeContext) {
```

Comment:
```suggestion
  const isInAIJSX = maybeContext && 'render' in maybeContext;
  if (isInAIJSX) {
```

minor, but would make the code more explicit for people will less familiarity